static/src/javascripts/projects/commercial/modules/hosted/video.js

```diff
@@ -60,84 +60,90 @@ define([
     function init(moduleName) {
         performanceLogging.moduleStart(moduleName);
 
-        require(['bootstraps/enhanced/media/main'], function() {
-            require(['bootstraps/enhanced/media/video-player'], function (videojs) {
-                var $videoEl = $('.vjs-hosted__video');
-                var $inlineVideoEl = $('video');
-                var $youtubeIframe = $('.js-hosted-youtube-video');
-
-                if ($youtubeIframe.length === 0 && $videoEl.length === 0) {
-                    if ($inlineVideoEl.length === 0) {
-                        // halt execution
-                        return;
-                    } else {
-                        $videoEl = $inlineVideoEl;
-                    }
-                }
-
-                $videoEl.each(function(el){
-                    var mediaId = $videoEl.attr('data-media-id');
-                    player = videojs(el, videojsOptions());
-                    player.guMediaType = 'video';
-                    videojs.plugin('fullscreener', fullscreener);
-
-                    events.addContentEvents(player, mediaId, player.guMediaType);
-                    events.bindGoogleAnalyticsEvents(player, window.location.pathname);
-
-                    player.ready(function () {
-                        var vol;
-                        var player = this;
-                        initLoadingSpinner(player);
-                        upgradeVideoPlayerAccessibility(player);
-
-                        // unglitching the volume on first load
-                        vol = player.volume();
-                        if (vol) {
-                            player.volume(0);
-                            player.volume(vol);
+        // Return a promise that resolves after the async work is done.
+        var enhanceVideo = new Promise(function(resolve){
+
+            require(['bootstraps/enhanced/media/main'], function() {
+                require(['bootstraps/enhanced/media/video-player'], function (videojs) {
+                    var $videoEl = $('.vjs-hosted__video');
+                    var $inlineVideoEl = $('video');
+                    var $youtubeIframe = $('.js-hosted-youtube-video');
+
+                    if ($youtubeIframe.length === 0 && $videoEl.length === 0) {
+                        if ($inlineVideoEl.length === 0) {
+                            // halt execution
+                            performanceLogging.moduleEnd(moduleName);
+                            resolve();
```

Comment:
`resolve` does not stop execution, that is why there needs to be a `return` statement.